tests/Unit/IO/H5/Test_H5.cpp

```diff
@@ -205,6 +206,62 @@ void test_errors() {
     file_system::rm("Unit.IO.H5.read_data_error.h5", true);
   }
 }
+
+void test_list_files() {
+  const std::string h5_file_name("Unit.IO.H5.list_files.h5");
+  if (file_system::check_if_file_exists(h5_file_name)) {
+    file_system::rm(h5_file_name, true);
+  }
+  // Insert a plethora of files to test listing with. Data doesn't matter
+  {
+    h5::H5File<h5::AccessType::ReadWrite> my_file(h5_file_name);
+    const std::vector<std::string> legend{{"Fake"}};
+    my_file.insert<h5::Dat>("/root_dat", legend, 0);
+    my_file.close_current_object();
+    my_file.insert<h5::Dat>("/group0/next_dat", legend, 0);
+    my_file.close_current_object();
+    my_file.insert<h5::Dat>("/group0/other_dat", legend, 0);
+    my_file.close_current_object();
+    my_file.insert<h5::VolumeData>("/root_vol_1", 0);
+    my_file.close_current_object();
+    my_file.insert<h5::VolumeData>("/root_vol_2", 0);
+    my_file.close_current_object();
+    my_file.insert<h5::VolumeData>("/group0/sub_vol", 0);
+  }
+
+  // Expected quantities
```

Comment:
I'm not sure HDF5 provides any guarantees on group/file ordering. It might be worth either using `std::unordered_set` and comparing that, or adding a note to this comment that `If ordering is an issue, switch to std::unordered_set because HDF5 may not guarantee file ordering`

Reply:
I've always seen that our files and groups are in alphabetical order, regardless of when they were inserted.

Reply:
But that ordering is defined by HDF5 and I'm not sure it's guaranteed :shrug: I don't think a comment would hurt